### PR TITLE
New analysis tests

### PR DESCRIPTION
**Pull request by mayabrandi**

### CODE REVIEW

File: tests/build/test_build_analysis.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from vogue.build.analysis import build_analysis
+from vogue.models.analysis import ANALYSIS_SETS
+from vogue.tools.cli_utils import json_read
+
+INVALID_JSON = 'tests/fixtures/not_a_multiqc_report.json'
+VALID_JSON = 'tests/fixtures/valid_multiqc.json'
+NO_FILE = 'tests/fixtures/no_exist_file.data'
+NOT_FILE = 'tests/fixtures'
```
Comment (ingkebil) [inline tests/build/test_build_analysis.py]: The above 4 constants could be moved to to conftest and set up as fixtures.

File: tests/build/test_build_analysis.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import pytest	
+
+
+def test_build_analysis():
+
+    ## GIVEN a analysis dictionary valid_analysis_dict and analysis_type
+    ## 1. read json file json_read
```
Comment (ingkebil) [inline tests/build/test_build_analysis.py]:
```
# are used for comments
## are used for "completely ignore this"
```
I don't think there is any reason to use double quotes, these are relevant comments.

(I'm just pointing this out, feel free to leave it as is)

File: tests/cli/load/test_load_analysis.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+from vogue.server import create_app
+from vogue.commands.base import cli
+from vogue.adapter.plugin import VougeAdapter
+
+VALID_JSON = 'tests/fixtures/valid_multiqc.json'
+INVALID_JSON = 'tests/fixtures/not_a_multiqc_report.json'
```
Comment (ingkebil) [inline tests/cli/load/test_load_analysis.py]: These constants could be moved to conftest as fixtures

File: tests/cli/load/test_load_analysis.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+app = create_app(test= True)
+
+
+def test_analysis(database):
+    app.db = database
+    app.adapter = VougeAdapter(database.client, db_name = database.name)
+
+    ## GIVEN a correct foramted input file VALID_JSON
```
Comment (ingkebil) [inline tests/cli/load/test_load_analysis.py]:
```suggestion
    ## GIVEN a correct formated input file VALID_JSON
```

File: tests/cli/load/test_load_analysis.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    sample_id = 'some_id'
+    
+    ## WHEN adding a new analysis
+    runner = app.test_cli_runner()
+    result = runner.invoke(cli, ['load', 'analysis', '-s', sample_id ,'-a', VALID_JSON ,'-t', 'QC' ])
+
+    ## THEN assert the new apptag should be added to the colleciton
+    assert isinstance(app.adapter.analysis(sample_id), dict)
```
Comment (ingkebil) [inline tests/cli/load/test_load_analysis.py]: Is an empty dict also valid?

File: tests/cli/load/test_load_analysis.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+def test_analysis_no_file(database):
+    app.db = database
+    app.adapter = VougeAdapter(database.client, db_name = database.name)
+
+    ## GIVEN a invalid path to a json file
+    sample_id = 'some_id'
+    
+    ## WHEN adding a new analysis
+    runner = app.test_cli_runner()
+    result = runner.invoke(cli, ['load', 'analysis', '-s', sample_id ,'-a', 'path' ,'-t', 'QC' ])
```
Comment (ingkebil) [inline tests/cli/load/test_load_analysis.py]: Can you move the 'path' into a variable just like `sample_id` is now a variable?

File: tests/parse/test_parse_analysis.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    ## 1. read json file json_read
+    ## 2. validate input dict vid the analysis set models
+
+    analysis_conf = json_read(VALID_JSON)
+
+    ## WHEN extracting the valid keys from the json, using validate_conf
+    mongo_sample_analysis = validate_conf(analysis_conf = analysis_conf)
+
+    ## THEN the putput should be a list and not a empty list
+    assert isinstance(mongo_sample_analysis, list)
```
Comment (ingkebil) [inline tests/parse/test_parse_analysis.py]: The comment says the test is testing that the output is not an empty list. The assert however does not do that.

File: tests/parse/test_parse_analysis.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+def test_validate_conf():
+
+    ## GIVEN a valid json file with some valid analysis keys
+    ## 1. read json file json_read
+    ## 2. validate input dict vid the analysis set models
+
+    analysis_conf = json_read(VALID_JSON)
+
+    ## WHEN extracting the valid keys from the json, using validate_conf
+    mongo_sample_analysis = validate_conf(analysis_conf = analysis_conf)
```
Comment (ingkebil) [inline tests/parse/test_parse_analysis.py]: is the result here a `mongo_sample_analysis`?

**Comment by ingkebil** [comment]
Also, put this code through a linter so that PEP8 can give you some useful comments :)

```
git checkout new_analysis_tests
git reset --soft 979b0453b6fb9548537724d4ffbe46a90e23e562
git lint
```

In cg we do something like this for automatic linting: https://github.com/Clinical-Genomics/cg/blob/master/.travis.yml#L15

**Comment by hassanfa** [comment]
> Looking good!
> 
> These are new test right? Not replacing previous unittests?

They are replacing some, and adding some new tests.

File: tests/cli/load/test_load_analysis.py (hunk 5 above)
Comment (hassanfa) [reply tests/cli/load/test_load_analysis.py]: It is. It is unclear here, probably needs to be explained better. I'll look into it.

File: tests/parse/test_parse_analysis.py (hunk 7 above)
Comment (hassanfa) [reply tests/parse/test_parse_analysis.py]: Comment is wrong. Only a list is acceptable for a valid input file. Needs one more assert of None.

File: tests/parse/test_parse_analysis.py (hunk 8 above)
Comment (hassanfa) [reply tests/parse/test_parse_analysis.py]: No, bad variable naming. Needs to renamed properly 👍

File: tests/cli/load/test_load_analysis.py (hunk 5 above)
Comment (hassanfa) [reply tests/cli/load/test_load_analysis.py]: Any dictionary is valid. I'll make an issue to create a test for empty dictionary and its consequences.

**Comment by hassanfa** [comment]
tests addded and fixtures updated. Needs an approval to merge.

**Comment by ingkebil** [comment]
Looking good!

Don't forget to run a linter on your code at some point.

**Pull request merged**